indispensable-asn1/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Real.kt

```diff
@@ -0,0 +1,182 @@
+package at.asitplus.signum.indispensable.asn1
+
+import at.asitplus.KmmResult
+import at.asitplus.catching
+import at.asitplus.signum.indispensable.asn1.encoding.decodeToAsn1Real
+import at.asitplus.signum.indispensable.asn1.encoding.encodeToAsn1Primitive
+import at.asitplus.signum.indispensable.asn1.encoding.fromTwosComplementByteArray
+import at.asitplus.signum.indispensable.asn1.encoding.toTwosComplementByteArray
+import at.asitplus.signum.indispensable.asn1.encoding.toUnsignedByteArray
+import kotlin.math.pow
+import kotlin.math.sign
+
+
+private const val IEEE754_BIAS = 1023
+
+/**
+ * ASN.1 REAL number. Mind possible loss of precision compared to Kotlin's built-in types.
+ * This type is irrelevant for PKI applications, but required for generic ASN.1 serialization
+ */
+sealed class Asn1Real : Asn1Encodable<Asn1Primitive> {
+
+    /**
+     * Converts this Asn1Real to a [Float]. **Beware of *probable* loss of precision and the fact that ANS.1 REAL zero knows no sign!**
```

Comment:
```suggestion
     * Converts this Asn1Real to a [Float]. **Beware of *probable* loss of precision and the fact that ASN.1 REAL zero knows no sign!**
```

Reply:
also why is the zero thing a problem? isn't the zero thing only a problem for going float/double to real, not the other way?